src/ScriptedInputSource.cpp

```diff
@@ -84,6 +84,40 @@ PlayerInputAbs ScriptedInputSource::getNextInput()
 		state.swapSides();
 	}
 
+	// if the ball is on the bots side, decrease the error timers more quickly.
+	// if we did not do this, then the difficulty would have almost no effect on
+	// any plays made by the player from the back of their half, as the counters
+	// would have reached zero by the time the ball arrives at the bot's half. This
+	// way, we can give the counters larger initial values, and still don't get completely
+	// stupid play.
+	if(state.getBallPosition().x < 400) {
+		--mBallPosErrorTimer;
+		--mReactionTime;
+	}
+
+	// decrement counters regularly
+	--mReactionTime;
+	--mBallPosErrorTimer;
+
+	// bot has not reacted yet
+	if( mReactionTime > 0 && mDifficulty > 0 ) {
```

Comment:
It is possible mReactionTime is 0 or lower. When opponent serves this can block the game.